src/wp-includes/post.php

```diff
@@ -2289,7 +2296,19 @@ function add_post_type_support( $post_type, $feature, ...$args ) {
 	$features = (array) $feature;
 	foreach ( $features as $feature ) {
 		if ( $args ) {
-			$_wp_post_type_features[ $post_type ][ $feature ] = $args;
+			// Check if feature already exists with args and if both are associative arrays that should be merged.
+			if ( isset( $_wp_post_type_features[ $post_type ][ $feature ][0] )
+				&& is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] )
```

Comment:
Per below, should this check to make sure it is actually an associative array? I think it should add this check: `! array_is_list( $_wp_post_type_features[ $post_type ][ $feature ][0] )`